source/jquery.flot.hover.js

```diff
@@ -167,9 +167,33 @@ the tooltip from webcharts).
 
             pos.pageX = page.X;
             pos.pageY = page.Y;
-
-            var item = plot.findNearbyItem(canvasX, canvasY, seriesFilter, distance);
-
+            
+            var items = [];
+            var item = undefined;
+            var found = plot.findNearbyItem(canvasX, canvasY, seriesFilter, distance);
+            if (found){
+                items.push(found);
+                item = found;
+            }
+            
+            for (var i = 0; i < plot.hooks.clickHoverFindNearby.length; ++i){
```

Comment:
One odd thing about this implementation, is putting in the processing of hooks, made available through flot, from a plugin. Additionally, we've hidden what could be a nice addition to the public API through a narrow interactive use-case (finding more than one item).

It seems reasonable for us to add a new public method like 'findNearbyItems' (I think the similarity in name to the other method isn't awful) to jquery.flot, and then have its default implementation call the new hooks (through the standard executeHooks mechanism).

I think in order to do this you would have to pass the 'items' through the hook args, so any hook implementation would add found items to the passed in list.

Does this make sense?

Reply:
That seems reasonable. Seems like it should be a rather simple change too. I'll plan to start digging into it later.

Reply:
Would we want findNearbyItem changed to return the closest item found by findNearbyItems? (Moving the existing code into something else which would be interacted with like the rest of the hooks)

Reply:
This makes sense to me.

Reply:
I've pushed out an initial set of changes implementing this. I'm not overly comfortable with JS, so there's probably more work to make the change actually make sense.

Probably the biggest issue, which I really didn't want to mess with is that the base code for finding flot items still will only ever return 1 item, the closest. 

I missed documentation for findNearbyItems, so I'll have at least one more commit handling this that I can tack any requested changes onto.

Reply:
Sorry it's taking me awhile to address these changes. I'm interested in discussing this with some developers on my end, but haven't been able to pin them down. Basically, I share similar concerns to your own, and am wondering if there wasn't a cleaner way to achieve the functionality you need. Part of the expected conversation I intend to have is whether we need to add a new hook at all.

Some of my confusion stems from not fully understanding how you're leveraging the new hook. The public methods `findNearbyItem` and your new `findNearbyItems`, as they are, are expected to find values within a given circular radius. You mentioned wanting to get hover support to labels, but I'm not entirely clear how your change accomplishes this. All items that are going to be returned by these methods, including the new hook, represent series values, yes?

All I can see the new hook really doing is discovering "highlightable" points that found through a custom algorithm (non-radial), but that doesn't seem to gel with the stated intent. If this is true, I would think we could come up with a better way than a new hook to provide a custom search routine for the hover plugin.

Again, sorry for the delay on addressing this, and any clarity you can provide related to the above concerns would be appreciated.

Reply:
So, a bit of clarity to what we have implemented on our end. We're working with a plugin which adds labels to our graphs for each data point and have been extending it as necessary. So currently the plugin acts as if hovering over the label is exactly the same as hovering over the item in a series it is related to.

My primary intention with this was to allow plugins to extend what could be considered "highlightable" to make it easier for any plugin which would draw on the canvas to provide equivalent functionality to what is in base flot. Whether this needs to be done with a hook or not, I don't know.

Reply:
I see, and I presume you would have 'options.grid.autoHighlight' turned off, then capture the clickHover event, and then highlight the label in your plugin. Is this correct?

Reply:
We have highlights turned off generally for this area. We're using the hover events to run tooltips on the graph